contracts/rewards/src/contract/execute.rs

```diff
@@ -156,18 +173,125 @@ mod test {
         }
     }
 
-    fn create_contract(stored_params: StoredParams) -> Contract<state::MockStore> {
+    /// Tests that rewards are added correctly to a single contract
```

Comment:
the test look good overall, but I'd like to further remove them from implementation details (e.g. the important thing is that the pool tracks the correct amount, not if it's initialized or not)
I think we need a query to check reward pool levels anyway, so that could be a good way to further remove the test from the exact data model (in a follow-up?). So that means instead of checking the store directly you could query for the amount

Reply:
We can add a query, and possibly other queries, in a follow up